### PR TITLE
[Material 3] Chips showCheckmark from theme being ignored

### LINKED ISSUE #119163: `RawChip` doesn't use `ChipThemeData.showCheckmark` value
## Use case

Hello 👋 
While using the new m3 chips, it looks like the RawChip widget itself ignores the `showCheckmark` from the themes, because its instance value (priority) has a default `= true` on the constructor.

So whenever I wanted to use a `RawChip(...)`, I would have to `RawChip(showCheckmark: null, ...)` instead, so it fallbacks to the theme value.

> My current flutter version (beta channel)
```
Flutter 3.7.0-1.4.pre • channel beta • git@github.com:flutter/flutter.git
Framework • revision 686fe913dc (2 weeks ago) • 2023-01-10 16:14:01 -0800
Engine • revision ac5c83dd5f
Tools • Dart 2.19.0 (build 2.19.0-444.4.beta) • DevTools 2.20.0
```

## Proposal

A simple proposal would be [removing the default constructor value](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/chip.dart#L758) since the property is already nullable.

Also, ChoiceChip particularly will always override the showCheckmark with the [`theme.isMaterial3`](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/choice_chip.dart#L159), rather than having it defined on its [internal default variation theme](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/choice_chip.dart#L197).

---

Not sure if this is better to be a separate issue but I would also love to be able to insert a [custom constraint property](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/chip.dart#L1288-L1308), either on the theme or instance.

### PR DESCRIPTION
**Pull request by GayathrideviGA**
M3 chips, it looks like the RawChip widget itself ignores the showCheckmark from the themes, because its instance value (priority) has a default = true on the constructor.

Previously, the checkmark in the ChoiceChip component was not being hidden when using Material 3. 

To address this issue, we have added two new parameters to the ChoiceChip component. The first parameter, showCheckMark, allows us to hide or show the checkmark based on true/false conditions. The second parameter, checkMarkColor, provides the ability to customize the colour of the checkmark displayed.
With these changes, the ChoiceChip component now works as expected with Material 3, and provides greater flexibility and control over the display of the checkmark.

Fixes #119163 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by rydmike** [comment]
Thanks @GayathrideviGA, this is a nice improvement. Please consider adding these capabilities to `Chip` theming as well. 

Theming should imo always be on par with widget styling capabilities and with what M3 defaults can deliver. Otherwise we negate the usefulness of theming as a concept and force users to the poor practice of using widget styling over making the theme match your general design need.

The `Chip` theming is sadly already a bit flawed and has a number of issues making it impossible to with themes reproduce the M3 styles, using e.g. different colors, see https://github.com/flutter/flutter/issues/115364

cc @HansMuller @guidezpl

**Comment by rydmike** [comment]
Maybe it is already in there, I might have missed it. The theme has checkMark and color for it, so if it already using them as fallback after new widget props and before M2/M3 defaults, it should be fine.

I forgot how messy the `Chip` variant implementations are. Considering that they still don't support correct M3 specs, and we cannot recreate the variants with theming, using different colors choices than you get via defaults. I'm still torn between if it was the right choice to re-use the old M2 widget as base for creating the M3 versions.

File: packages/flutter/test/material/choice_chip_test.dart [hunk 1]
```diff
@@ -80,7 +80,7 @@ void main() {
 
     await tester.pumpWidget(buildFrame(Brightness.light));
     expect(getMaterialBox(tester), paints..rrect(color: const Color(0x3d000000)));
-    expect(tester.getSize(find.byType(ChoiceChip)), const Size(108.0, 48.0));
+    expect(tester.getSize(find.byType(ChoiceChip)), const Size(132.0, 48.0));
```
Comment (HansMuller) [inline packages/flutter/test/material/choice_chip_test.dart:83]: Why did these two tests change? Were we not showing the checkmark before?

**Review by HansMuller — COMMENTED** [review]
This PR should include regression tests that demonstrate that one can configure both new parameters via the ChoiceChip widget and the chip theme.

**Comment by TahaTesser** [comment]
Hi @GayathrideviGA 
I just reviewed https://github.com/flutter/flutter/pull/124743 and it looks like it addresses the same issue and also contains the test and it was also filed before this PR.

Thank you for your contribution. Next time please confirm a PR is already open for an issue that addresses it. I'm closing this.

**Pull request closed without merging**